sdk/python/setup.py

```diff
@@ -46,6 +46,7 @@
     "numpy",
     "google",
     "confluent_kafka",
+    'boto3'
```

Comment:
I think we should remove both the GCP and AWS clients here and ask users to install them with descriptive exceptions. @jmelinav what do you think?

Reply:
This change will force an adoption for the existing users. is that okay?

Reply:
@woop I went ahead and removed only boto3, and left google cloud storage dependency. so that existing users are not affected.

Reply:
That's ok. I think we should phase out our GCP requirement over time thought.

[I quite like how Airflow does this](https://airflow.apache.org/docs/stable/installation.html)